crates/environ/src/component/types.rs

```diff
@@ -371,413 +368,308 @@ impl ComponentTypesBuilder {
-                    unimplemented!("gc types");
-                }
-                wasmparser::ModuleTypeDeclaration::Export { name, ty } => {
-                    let prev = result
-                        .exports
-                        .insert(name.to_string(), self.entity_type(ty)?);
-                    assert!(prev.is_none());
-                }
-                wasmparser::ModuleTypeDeclaration::Import(import) => {
-                    let prev = result.imports.insert(
-                        (import.module.to_string(), import.name.to_string()),
-                        self.entity_type(&import.ty)?,
-                    );
-                    assert!(prev.is_none());
-                }
-                wasmparser::ModuleTypeDeclaration::OuterAlias {
-                    kind: wasmparser::OuterAliasKind::Type,
-                    count,
-                    index,
-                } => {
-                    let ty = self.core_outer_type(*count, TypeIndex::from_u32(*index));
-                    self.push_core_typedef(ty);
-                }
+        types: types::TypesRef<'_>,
+        id: types::TypeId,
+    ) -> Result<TypeDef> {
+        Ok(match types.type_from_id(id).unwrap() {
+            types::Type::Defined(_) => TypeDef::Interface(self.defined_type(types, id)?),
+            types::Type::Module(_) => TypeDef::Module(self.convert_module(types, id)?),
+            types::Type::Component(_) => TypeDef::Component(self.convert_component(types, id)?),
+            types::Type::ComponentInstance(_) => {
+                TypeDef::ComponentInstance(self.convert_instance(types, id)?)
             }
-        }
-
-        self.pop_type_scope();
-
-        Ok(self.component_types.modules.push(result))
-    }
-
-    fn entity_type(&self, ty: &wasmparser::TypeRef) -> Result<EntityType> {
-        Ok(match ty {
-            wasmparser::TypeRef::Func(idx) => {
-                let idx = TypeIndex::from_u32(*idx);
-                match self.core_outer_type(0, idx) {
-                    TypeDef::CoreFunc(idx) => EntityType::Function(idx),
-                    _ => unreachable!(), // not possible with valid components
-                }
+            types::Type::ComponentFunc(f) => {
+                TypeDef::ComponentFunc(self.convert_component_func_type(types, f)?)
             }
-            wasmparser::TypeRef::Table(ty) => EntityType::Table(self.convert_table_type(ty)),
-            wasmparser::TypeRef::Memory(ty) => EntityType::Memory(ty.clone().into()),
-            wasmparser::TypeRef::Global(ty) => EntityType::Global(self.convert_global_type(ty)),
-            wasmparser::TypeRef::Tag(_) => bail!("exceptions proposal not implemented"),
+            types::Type::Instance(_) | types::Type::Func(_) | types::Type::Array(_) => {
+                unreachable!()
+            }
+            types::Type::Resource(_) | types::Type::Cont(_)  => unimplemented!(),
```

Comment:
This file saw a major refactoring upstream (see 2e56c973c, wasmtime PR no. 6638)

I've added this `unimplemented` which now applies more broadly than a similar one that was in this file before the merge, which only affected `wasmparser::ModuleTypeDeclaration::Type(wasmparser::Type::Cont(_))`.

The tests pass and I guess we will find out easily if this ever needs to change.

Reply:
It's fine. We largely ignore all component-related stuff for now as it is very much in flux.